File: sdk-py/noether_sdk/sub/health.py
from ..models import HealthStatus
from ..transport import Transport


class HealthApi:
    def __init__(self, transport: Transport) -> None:
        self._transport = transport

    async def ping(self) -> HealthStatus:
        body = await self._transport.request("GET", "/v1/health")
        return HealthStatus.model_validate(body)

File: sdk-ts/src/sub/health.ts
import type { Transport } from '../transport.js';

export interface HealthStatus {
  status: string;
  uptime: number;
  version: string;
}

export class HealthApi {
  constructor(private readonly transport: Transport) {}

  async ping(): Promise<HealthStatus> {
    return this.transport.request<HealthStatus>({ path: '/v1/health' });
  }
}
